### ISSUE TITLE
init command produces various error messages, cannot complete

### ISSUE THREAD

**Issue by mattwelke** [CircleCI-Public/circleci-cli#495]
This is probably the worst GitHub issue title I've ever written. :P The reason is because there were multiple issues I encountered. I'll do my best to describe them below. Someone with more context on CircleCI is free to rename this issue if they would like to.

## Meta:

**CircleCI CLI Version:**
0.1.10993+cb3ba09 (release)

**Operating System:**
Ubuntu 19.10

**CircleCI CLI Diagnostic:**
---
CircleCI CLI Diagnostics
---
Debugger mode: false
Config found: /home/matt/.circleci/cli.yml
API host: https://circleci.com
API endpoint: graphql-unstable
OK, got a token.
Trying an introspection query on API... 
Ok.
Hello, Matt Welke.

---

## Current behavior:
Interactive prompts cease after I enter my GitHub repo URL and hit enter, at which point an error is also displayed:

```
> circleci orb init code/github.com/mattwelke/circleci-orb-test
Note: This command is in preview. Please report any bugs! https://github.com/CircleCI-Public/circleci-cli/issues/new/choose
✔ Yes, walk me through the process.
Downloading Orb Project Template into code/github.com/mattwelke/circleci-orb-test
A few questions to get you up and running.
✔ GitHub
✔ Enter your github username or organization: mattwelke
✔ Enter the namespace to use for this orb: mattwelke
Saving namespace mattwelke as default
✔ Orb name: circleci-orb-test
✔ Yes, set up a publishing context with my API key.
✔ Yes, set up the git project.
✔ Enter the remote git repository: git@github.com:mattwelke/circleci-orb-test.gi✔ Enter the remote git repository: git@github.com:mattwelke/circleci-orb-test.gi✔ Enter the remote git repository: git@github.com:mattwelke/circleci-orb-test.git
Thank you! Setting up your orb...
Error: author field is required
```

And, when I delete the directory that was created and then try the init command again, it stops at the same spot, and a different error is displayed:

```
~ > rm -r code/github.com/mattwelke/circleci-orb-test
~ > circleci orb init code/github.com/mattwelke/circleci-orb-test
Note: This command is in preview. Please report any bugs! https://github.com/CircleCI-Public/circleci-cli/issues/new/choose
✔ Yes, walk me through the process.
Downloading Orb Project Template into code/github.com/mattwelke/circleci-orb-test
A few questions to get you up and running.
✔ GitHub
✔ Enter your github username or organization: mattwelke
✔ Enter the namespace to use for this orb: mattwelke
Saving namespace mattwelke as default
✔ Orb name: circleci-orb-test
✔ Yes, set up a publishing context with my API key.
✔ Yes, set up the git project.
✔ Enter the remote git repository: git@github.com:mattwelke/circleci-orb-test.git
Thank you! Setting up your orb...
Error: Non-nullable field was null.
A context named orb-publishing already exists.
```

And when I try to finish it manually, reading the docs, by running `git init`, `git remote add origin <url>`, `git add`, `git commit`, and then `git push origin master`, following the project manually on the CircleCI UI (clicking "start building"), the CircleCI UI shows an error saying it can't build it:

![image](https://user-images.githubusercontent.com/7719209/96376165-41daa680-114b-11eb-88ad-6b1e01512b8d.png)

## Expected behavior:
No errors are output during the interactive steps. If an error occurs, the interactive process is able to be run from the beginning again cleanly.

## When did this begin / Was this previously working?:
This have never been working for me. This is my first experience with authoring CircleCI orbs.

**Comment by gmemstr** [comment]
Interesting error `Error: author field is required` - taking a deeper look into this, I suspect it's related to the git library we're using for the CLI and its inability to pull in a git identity. As for the 
```
Error: Non-nullable field was null.
A context named orb-publishing already exists.
```
error, that's pretty standard, although maybe something we should handle/communicate better. The TL;DR is that the first time an orb is initialised and the `Yes, set up a publishing context with my API key.` option is selected, we create the context, and any orbs initialised in the future should opt to not create this context.

**Comment by gmemstr** [comment]
@mattwelke Curious if you installed the CLI via `snap`. Trying to narrow down possibilities.

**Comment by mattwelke** [comment]
I tried first with the snap, and it didn't even get this far. With the snap, I encountered https://github.com/CircleCI-Public/circleci-cli/issues/491. Switching to the binary let me get past that error, so that I could progress this far and encounter this error.

**Comment by mattwelke** [comment]
I'm happy to test again because I'll be working mostly on CircleCI and orbs this week. Just let me know.

**Comment by gmemstr** [comment]
Second error is being taken care of in #503 - just need better messaging/handling when the context already exists. Ideally one would pick "no" if it already existed, but it's possible it's not known whether it already exists.

**Comment by Jean-Baptiste-Lasselle** [comment]
Hi @mattwelke Did you not experience this new issue : https://github.com/CircleCI-Public/circleci-cli/issues/491#issuecomment-722531733 ?

**Comment by mattwelke** [comment]
@Jean-Baptiste-Lasselle 

I actually finished up with my CircleCI work early and so I haven't revisited this since originally opening this issue. I haven't run any commands to see if there are any new errors produced. It looks like you already reported it in that issue, so I'd say that's all you can do for now.

**Comment by shay-ul** [comment]
Hey @gmemstr , looks like I'm experiencing this issue as well.
I have double checked and it looks like my `git config user.email` is set globally.
Is there a way to troubleshoot this behavior?
Do we have a workaround to complete the initialisation of the orb even when encountering such error?

**Comment by Peter-Darton-i2** [comment]
I've just encountered this.
Firstly by when I assumed the tool would cope with me setting up the folder it was working in (failed because the local git repo already existed - see #803)
Secondly when I tried following the instructions slavishly and got the "author field is required" error (i.e. as reported here)
Thirdly when I told it not to mess with the git repo and it failed because the orb had already been registered.

There seems to be a general pattern of the "orb init" code unsafely assuming that all resources it wants to exist don't already exist. That means that it can only ever be run in a totally clean environment, that it can never be re-run etc.
